### PR TITLE
feat(about-modal): add help dropdown and about modal

### PR DESCRIPTION
**Pull request by seanforyou23**
This PR adds a help dropdown to the masthead with a single item that launches an about modal.

Helps address https://github.com/syndesisio/syndesis-react/issues/197

With Trademark Text "Red Hat"
Screen Shot 2019-05-01 at 8 41 19 PM

Without Trademark Text
Screen Shot 2019-05-02 at 3 56 25 PM

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/923/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/941/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by mcoker** [comment]
The "A flexible, customizable, open source platform that provides core integration capabilities as a service." text is missing `.pf-m-md` on the `.pf-c-title` element that wraps the text. I see that class here - http://patternfly-react.surge.sh/patternfly-4/components/aboutmodal/. Not sure why it isn't on your version. It's creating to small of a line-height.

You should have a `.pf-c-backdrop` element that dims the content behind the modal. Looks like this bit of CSS is being applied after the backdrop component, overwriting the backdrop's background-color?

```
[class*=pf-c-], [class*=pf-c-]:after, [class*=pf-c-]:before, [class*=pf-l-], [class*=pf-l-]:after, [class*=pf-l-]:before {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
    background-color: #0000;
}
```

You might set something like `pf-c-about-modal-box__content dd { word-break: break-word; }` to allow the ID's to break onto a new line. I opened an issue in core here - https://github.com/patternfly/patternfly-next/issues/1784

At a certain breakpoint, I'm seeing this black band along the right of the modal. At a glance I can't tell what's causing that. Looks like it might have something to do with `--pf-c-about-modal-box--lg--grid-template-columns` but I'm not seeing it on the [pf4 react example](http://patternfly-react.surge.sh/patternfly-4/components/aboutmodal/)?

File: app/ui-react/syndesis/src/app/App.tsx [hunk 1]
```diff
@@ -201,6 +211,31 @@ export class App extends React.Component<IAppBaseProps, IAppBaseState> {
                           <WithServerEvents apiUri={apiUri} headers={headers}>
                             {functions => (
                               <ServerEventsContext.Provider value={functions}>
+                                <WithApiVersion>
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/app/App.tsx]: Does this poll?  If it does, then it'll be polling all the time, and if it doesn't it'll run when the page loads up.

File: app/ui-react/packages/ui/src/Shared/SynAboutModal.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      <React.Fragment>
+        <AboutModal
+          isOpen={isModalOpen}
+          onClose={handleModalToggle}
+          trademark={trademark}
+          brandImageSrc={brandImg}
+          brandImageAlt="Brand Image"
+          productName={productName}
+        >
+          <AboutModalContent
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Shared/SynAboutModal.tsx]: If you don't put this here and instead just have a `children` prop then you can organize the modal like I had mentioned, where you can do the API call from within the body of the modal, so only when it's shown vs when the page is loaded

File: app/ui-react/syndesis/src/app/App.tsx (hunk 1 above)
Comment (gashcrumb) [reply app/ui-react/syndesis/src/app/App.tsx]: You could also probably do `{this.state.showAboutModal && (<WithApiVersion>` here actually, that'd mean the call would be made when the modal is rendered on the page.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/951/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/syndesis/src/app/App.tsx [hunk 3]
```diff
@@ -199,56 +210,87 @@ export class App extends React.Component<IAppBaseProps, IAppBaseState> {
...
+                              return (
+                                <ServerEventsContext.Provider value={functions}>
+                                  <AboutModal
+                                    trademark={'Red Hat'}
+                                    productName={productName}
+                                    isModalOpen={this.state.showAboutModal}
+                                    handleModalToggle={this.toggleAboutModal}
+                                    brandImg="https://avatars0.githubusercontent.com/u/23079786"
+                                  >
+                                    {this.state.showAboutModal && (
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/app/App.tsx]: This line should go up in front of the `<AboutModal>` tag there on line 217 I think.

**Comment by gashcrumb** [comment]
It's almost there!  I only spotted one thing.  Did you want to do the `WithLoader` stuff in this PR tomorrow?

**Comment by seanforyou23** [comment]
Yes that would be great, I've been chipping away but the loader/error states are important as I'll need to know how to properly incorporate that for a lot of other features that are in the backlog. Learning a lot on with this one, let's take it all the way with loaders and all tomorrow.

**Comment by seanforyou23** [comment]
@dongniwang @amysueg see updated description, I've been searching for the trademark text but haven't found anyone who knows what exactly to use for copy there. Should we just remove it for now? This example in PF doesn't have that information so maybe we're fine without? http://patternfly-react.surge.sh/patternfly-4/components/aboutmodal/

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/953/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
Hey, sorry for the conflicts, I just pushed some changes that we needed to make the app more resilient on FF. The `App` is now split into two components, `App` that sets up the `AppContext`,  and `UI` that sets up the `UIContext` and feeds the `AppLayout` the props it needs.

You'll want to move the about modal in the `UI` component, and sync the `AppLayout` changes to the new function component (this was done to fix #206).

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/976/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by dongniwang** [comment]
Great work @seanforyou23! 

A few comments:
- the font looks a lot bigger than the regular body text we have in the app, maybe we can tune it down a little bit? WDYT? @mcoker @lboehling 
- need the latest background image 
- For the trademark section
   - upstream version, we can just leave it as an empty string for now 
   - for the productized version, @TovaCohen @gaughan do you know if we have placeholder text for that? Or if we should create an issue in JIRA to track that?

**Comment by gaughan** [comment]
@dongniwang probably need a Jira

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/987/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/989/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by dongniwang** [comment]
@seanforyou23 - do you want to keep the initial screenshot with the trademark text so we can keep track of the decision?

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1001/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Pull request merged**